site/content/en/docs/tasks/integrate_a_custom_job.md

```diff
@@ -0,0 +1,61 @@
+---
+title: "Integrate a custom Job"
+date: 2023-07-25
+weight: 8
+description: >
+  Integrate a custom Job with Kueue.
+---
+# Integrate a custom Job with Kueue
+
+Kueue integrates with a couple of jobs, including batch Job, MPI Job, Ray Job or JobSet. 
+If there is another Job, which you would like supported, please consider contributing. 
+This page shows how to integrate a custom Job with Kueue.
+
+## What you need to specify
+
+1. Configuration
+2. Controller
+3. Webhook
+4. Adjust build system
+5. Test files
+
+## Configure your custom Job
+
+Add your framework name to `.integrations.frameworks` in [controller_manager_config.yaml](https://github.com/kubernetes-sigs/kueue/blob/main/config/components/manager/controller_manager_config.yaml)
+
+## Controller
+
+Provide your code of the controller under a dedicated directory. This directory need to under the `./pkg/controller/jobs/` folder.
+
+1. Register a new framework
+2. Add RBAC Authorization
+    - Add RBAC Authorization for priorityclasses, events, workloads, resourceflavors and any resources you want.
+3. Implement interfaces
```

Comment:
I think the `GenericJob` is particularly important so it should be called here. Maybe
```
3. Implement the `GenericJob` interface, and potentially other interfaces in [here](https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobframework/interface.go).
```